geom/type_line.go

```diff
@@ -172,6 +172,10 @@ func (n Line) Length() float64 {
 	return math.Sqrt(delta.Dot(delta))
 }
 
+func (n Line) Centroid() Point {
+	return NewPointF((n.a.XY.X+n.b.XY.X)/2, (n.a.XY.Y+n.b.XY.Y)/2)
```

Comment:
An alternate way to do this could be `return NewPointXY(n.a.Add(n.b.XY).Scale(0.5))`. Something to consider if you didn't see that those helpers existed (if you disagree that this makes it easier to read, then it's probably subjective so feel free to ignore my comment).

Reply:
I wish Go had operator methods to allow mathematical operations on objects like points.